app/workers/certificates.php

```diff
@@ -26,8 +34,63 @@ public function init(): void
 
     public function run(): void
     {
+        Authorization::disable();
+
+        $type = $this->args['type'] ?? '';
+        switch (strval($type)) {
+            case CERTIFICATE_TYPE_RENEW:
+                $this->renewCertificate();
+                break;
+            case CERTIFICATE_TYPE_ISSUE:
+                $this->issueCertificate($this->args);
+                break;
+        }
+
+        Authorization::reset();
+    }
+
+    // This is recursive function
+    protected function renewCertificate() {
+        // If this occurs faster than cycle interval, it's left-over from previous container after restart. We ignore such an instance.
+        if(isset(self::$lastRenewCycle) && self::$lastRenewCycle + 60 > \time()) {
+            Console::log('Received left-over renew cycle event; Ignoring.');
+            return;
+        }
+        self::$lastRenewCycle = \time();
+
         $dbForConsole = $this->getConsoleDB();
 
+        // Check if we hit API limit
+        $certificatesCreated = $dbForConsole->sum('certificates', 'attempts', [
+            new Query('updated', Query::TYPE_LESSEREQUAL, [ \time() - 10800 ]) // 3 hours
+        ], 250);
+
+        if($certificatesCreated >= 250) { // Limit is 300, but we keep 50 to stay safe with new cert generation
+            Console::log('Renew cycle skipped due to potentialy exceeded API limits.');
+        } else {
+            // We will renew all certs older than 60 days
+            // Refference: https://letsencrypt.org/docs/faq/
+            // "Our certificates are valid for 90 days. We recommend automatically renewing your certificates every 60 days."
+            $certificateForRenewal = $dbForConsole->findOne("certificates", [
+                new Query('attempts', Query::TYPE_LESSER, [540]), // 3 API limit reset cycles
+                new Query('result', Query::TYPE_EQUAL, ['done', 'letsEncryptError']),
+                new Query('updated', Query::TYPE_LESSEREQUAL, [ \time() - 5184000 ]), // - 60 days
+            ]);
+
+            if($certificateForRenewal) {
+                $this->issueCertificate([
+                    'domain' => $certificateForRenewal->getAttribute('domain'),
+                ]);
+            }
+        }
+
+        // When done, schedule again in 1 minute. This creates makes the function recursive.
+        ResqueScheduler::enqueueIn(60, Event::CERTIFICATES_QUEUE_NAME, Event::CERTIFICATES_CLASS_NAME, [
```

Comment:
The main idea of this task was to decouple us from the `ResqueScheduler` to avoid trusting Redis for persistency. This could also potentially lead to to multiple infinite loops in some edge cases. Also why should we trigger renewal every 1 minute? The idea was to check for renewals from the maintenance container every day for certificates that were last issues +60 days ago.